src/org/apache/cocoon/transformation/UnzipTransformer.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE src distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this src to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this src except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cocoon.transformation;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import org.apache.avalon.framework.parameters.Parameters;
+import org.apache.cocoon.ProcessingException;
+import org.apache.cocoon.environment.SourceResolver;
+import org.apache.cocoon.xml.AttributesImpl;
+import org.apache.commons.io.FilenameUtils;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+/**
+ * This transformer unzips Zip files.
+ * <p>
+ * It triggers for elements in the namespace "http://apache.org/cocoon/unzip/1.0".
+ * <p>
+ * Example XML input:
+ * <pre>
+ * {@code
+ *   <unzip:archive src="file:///tmp/my-temp-store/zipfile.zip" 
+ *      target="file:///tmp/my-unzips/zipfile" recursive="true"/>
+ * }
+ * </pre>
+ The @src specifies the file that should be created.
+ The @target attribute specifies the directory where the content of the zip
+  will be stored. If unspecified, the content will unzipped in a folder
+ with the target of the zip file, minus the '.zip' extension.
+ If the @recursive attribute is "true" then zip files in the @src zip
+ will also be unzipped.
+ <p>
+ * The result is
+ * <pre>
+ * {@code
+ *   <zip:result target="file:///tmp/my-unzips/zipfile">OK</zip:result>
+ * }
+ * </pre>
+ * or it throws an error.
+ *
```

Comment:
Dit lijkt me meer iets voor een Cocoon action dan een transformer, maar ach, cocoon is EOL.

Reply:
:-) Waarschijnlijk heb je gelijk, is het meer een action.